src/extension/handler/uri.ts

```diff
@@ -103,13 +105,25 @@ export class RunmeUriHandler implements UriHandler {
         fileToOpen: string
     ) {
         progress.report({ increment: 0, message: 'Cloning repository...' })
-        const terminal = window.createTerminal(`Runme Terminal #${NEXT_TERM_ID++}`, '/bin/sh')
+        const terminal = window.createTerminal(`Runme Terminal #${NEXT_TERM_ID++}`)
         terminal.show(true)
 
-        terminal.sendText(`git clone ${repository} ${targetDirUri.fsPath}`)
+        terminal.sendText(`git clone ${gitSSHUrlToHTTPS(repository)} ${targetDirUri.fsPath}`)
         const success = await new Promise<boolean>(
             (resolve) => waitForProjectCheckout(fileToOpen, targetDirUri.fsPath, repository, resolve))
 
+        /**
```

Comment:
I need clarification on why this is necessary... One can't check out a private git repo without authorization, no matter whether it's HTTPS or SSH. The only upshot HTTPS has over SSH is for public repos where GIT authentication (forgoes authorization) requires pubkeys to be set up is performed as a result. HTTPS only does authentication if authorization is required which is just a fundamental protocol difference that plays nicely with public repos.

Reply:
> The only upshot HTTPS has over SSH is for public repos where GIT authentication (forgoes authorization) requires pubkeys to be set up is performed as a result.

Exactly that. With this patch cloning the repo will also work for folks that don't have git set up. But as you said, might be problematic for folks that try to clone a private repo. I guess the question is, who will use this more:

- folks that don't have git setup
- folks that try to checkout a private repository

I hunch was the first but if you say you like to focus on latter more I can disregard the PR.

Another option is to allow setting another url query parameter to tag the git resource as private in which case we don't do this transition.

Reply:
> Exactly that. With this patch cloning the repo will also work for folks that don't have git set up. But as you said, might be problematic for folks that try to clone a private repo. I guess the question is, who will use this more:
> 
> * folks that don't have git setup
> * folks that try to checkout a private repository
> 
> I hunch was the first but if you say you like to focus on latter more I can disregard the PR.
> 
> Another option is to allow setting another url query parameter to tag the git resource as private in which case we don't do this transition.

As I said before, we want to support both but not "programagically". It's not the extension's place to figure out what git repo reference (git vs https) to use. That's up to the author of the link/badge to decide. So in essence, if GIT is provided in the URI, use GIT, if HTTPS is provided, use HTTPS (as long as they are valid respectively).

We should then add a paragraph to the feature's docs to explain what to look out for.

Reply:
Gotcha, I miss-understood then from the [original issue](https://github.com/stateful/vscode-runme/issues/258#issue-1606894447) which mainly asked to improve the UX for users without git setup. I can have VS Code print a helpful error message for these users then.